backend/de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/material/event/PurchaseCandidateRequestedHandler.java

```diff
@@ -148,6 +145,17 @@ public void handleEvent(@NonNull final PurchaseCandidateRequestedEvent event)
 		saveCandidateAndPostCreatedEvent(event, newPurchaseCandidate);
 	}
 
+	@Nullable
+	private ZonedDateTime getPurchaseDateOrderedOrNull(@NonNull final ZonedDateTime datePromised, @Nullable final ProductPlanningId productPlanningId)
+	{
+		if (productPlanningId == null)
+		{
+			return null;
+		}
+		final int leadTimeDays = productPlanningDAO.getById(productPlanningId).getLeadTimeDays();
+		return datePromised.minusDays(leadTimeDays);
```

Comment:
in previous version, "calendarNonBusinessDays.getPreviousBusinessDay(purchaseDatePromised, (int)leadTimeOffset.toDays())" was considered.... why not anymore?

Reply:
because C_BP_PurchaseSchedule has a calendar column, while PP_ProductPlanning does not. Also, PP_Order_Candidate/DD_Order_Candidate don't care about business days either.